spec/views/users/index.html.erb_spec.rb

```diff
@@ -1,5 +1,32 @@
 require 'rails_helper'
 
-RSpec.describe 'users/index.html.erb', type: :view do
-  pending "add some examples to (or delete) #{__FILE__}"
+RSpec.feature 'User Index Page', type: :feature do
+  before :each do
+    @user = User.create(id: 1, name: 'Marc', photo: 'https://media.gettyimages.com/id/1278139568/', bio: 'Tech',
+                        posts_counter: 5)
+  end
+
+  scenario 'Displaying user information' do
+    @user1 = User.create(name: 'Lilly', photo: 'user1.jpg', bio: 'teacher', posts_counter: 1)
+    @user2 = User.create(name: 'user2', photo: 'user2.jpg', bio: 'farmer', posts_counter: 2)
+
+    visit users_path
+
+    # Test: Display username of all other users
+    expect(page).to have_content(@user1.name)
+    expect(page).to have_content(@user2.name)
+
+    # Test: Display profile picture for each user
+    expect(page).to have_css("img[src*='user1.jpg']")
+    expect(page).to have_css("img[src*='user2.jpg']")
+
+    # Test: Display number of posts each user has written
+    expect(page).to have_content('Number of posts: 1')
+  end
+
+  scenario 'renders the user show page' do
+    visit '/'
+    click_link(@user.name, match: :first)
+    expect(page).to have_content(@user.name)
```

Comment:
- [ ] Kindly something more specific to the user show page, like the browser's current route, instead of the user's name. The user's name is rather too generic and common to be the expectation, it exists on this index page as well and does not prove the button takes the user to the show page.

Reply:
Done!